Lib/test/test_shlex.py

```diff
@@ -358,6 +382,46 @@ def testJoinRoundtrip(self):
                 resplit = shlex.split(joined)
                 self.assertEqual(split_command, resplit)
 
+    def testJoinBytes(self):
```

Comment:
Use the same dataset as for those used in strings, especially with quotation symbols or unsafe characters.